### ISSUE TITLE
Most free space is not filled

### ISSUE THREAD

**Issue by gitthepie** [Divested-Mobile/Extirpater#3]
On a Pixel 7 device, I found that roughly 80% of the free space remained unfilled. I was watching the free space and refreshing at regular intervals to estimate how much longer the erase would take. The device had around 100 GB of free space, and I estimate no more than 20 GB was filled during the overwrite.

I have 2 questions here. First, I know the readme says "The file table of the secondary drive will never really be filled". But I thought this line was either referring to wear leveling or just a small portion of the space; not 80% of the free space!

Also, if Extirpater crashes during the wipe, is there a way, without root access, to clean up whatever temporary file it left while it attempted the wipe?

**Comment by SkewedZeppelin** [comment]
The space shown in the progress bar should be capable of being erased. This may be due to some restriction on cache size per app.

>  is there a way, without root access, to clean up whatever temporary file it left while it attempted the wipe?

It automatically clears it on app start and erase finish. You could also simply uninstall the app.

**Comment by Moonstone3010** [comment]
Same here.  I have internal memory full at 67% and the maximum full value when Extirpater is running is 71%.  And that 71% comes when the progress bar is about 1/3, after that nothing more happens to the data in memory.

**Comment by gitthepie** [comment]
I just tested Extirpater on a much older device running Android 8. Extirpater worked perfectly. I also confirmed that the cache on the app filled up as the erase was happening, so I finally know how to clear the cache should there be a cache. If the app crashes, I can just go into the app storage screen and clear the cache, and that should be the end of any potential leftover files.

I like this app a lot, and it's a shame the app doesn't work correctly on my newer phone. Is there anything we can do to help developers fix the problem?

**Comment by SkewedZeppelin** [comment]
fwiw simply opening the app will always delete all previous run data, you don't have to manually clear anything.

as for the limitation, an easy workaround may be simply dropping targetSdk, not too sure

**Comment by gitthepie** [comment]
Are you by any chance referring to this?: https://developer.android.com/guide/topics/manifest/uses-sdk-element.html

If you do release a new version with the change, could you please post here? I might be able to help test it.

**Comment by marzagheddon** [comment]
So? does this work or not?

**Comment by gitthepie** [comment]
No, it doesn't work. The developer clearly hasn't made a serious effort to fix this problem despite having some ideas on the cause, even when I made direct offers to help with testing. The limitations listed in the readme are just boilerplate and don't reflect the inability to fill such a significant percentage of the free space. If you need an app to securely erase unallocated storage, look elsewhere. Being an open source app is no excuse to abandon development.
